packages/ihe-gateway-sdk/src/models/document-query.ts

```diff
@@ -1,26 +1,74 @@
-import { BaseRequest } from "./shared";
+import {
+  BaseRequest,
+  BaseResponse,
+  BaseErrorResponse,
+  DocumentReference,
+  XCAGateway,
+  XCPDPatientId,
+  xcpdPatientIdSchema,
+  baseRequestSchema,
+  codeSchema,
+  Code,
+  baseResponseSchema,
+  baseErrorResponseSchema,
+  xcaGatewaySchema,
+  documentReferenceSchema,
+} from "./shared";
+import * as z from "zod";
 
-export type Code = {
-  system: string;
-  code: string;
-};
-
-export type DateRange = {
-  dateFrom: string;
-  dateTo: string;
-};
+export const dateRangeSchema = z.object({
+  dateFrom: z.string(),
+  dateTo: z.string(),
+});
+export type DateRange = z.infer<typeof dateRangeSchema>;
 
-export type DocumentQueryRequest = BaseRequest & {
-  xcaHomeCommunityId: string;
-  xcpdPatientId: {
-    id: string;
-    system: string;
-  };
+export type DocumentQueryRequestOutgoing = BaseRequest & {
+  cxId: string;
+  xcpdPatientId: XCPDPatientId;
   patientId?: string;
-  xcaGateway: string;
+  gateway: XCAGateway;
   classCode?: Code[];
   practiceSettingCode?: Code[];
   facilityTypeCode?: Code[];
   documentCreationDate?: DateRange;
   serviceDate?: DateRange;
 };
+
+export const documentQueryResponseIncomingSchema = z.union([
+  z.intersection(
+    baseResponseSchema,
+    z.object({
+      documentReference: documentReferenceSchema.array(),
+      gateway: xcaGatewaySchema,
+    })
+  ),
+  baseErrorResponseSchema,
+]);
+export type DocumentQueryResponseIncoming = z.infer<typeof documentQueryResponseIncomingSchema>;
+
+export const documentQueryRequestIncomingSchema = z.intersection(
+  baseRequestSchema,
+  z.object({
+    xcpdPatientId: xcpdPatientIdSchema,
+    classCode: z.array(codeSchema).optional(),
+    practiceSettingCode: z.array(codeSchema).optional(),
+    facilityTypeCode: z.array(codeSchema).optional(),
+    documentCreationDate: dateRangeSchema.optional(),
+    serviceDate: dateRangeSchema.optional(),
+  })
+);
+
+export type DocumentQueryRequestIncoming = z.infer<typeof documentQueryRequestIncomingSchema>;
+
+export type DocumentQueryResponseOutgoing =
+  | (BaseResponse & {
+      documentReference: DocumentReference[];
+    })
+  | BaseErrorResponse;
+
+export function isDocumentQueryResponse(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  obj: any
```

Comment:
NABD, but it seems we could define the type here as `BaseResponse` and get rid of the ESLint directive